custom_components/sscpoe/config_flow.py

```diff
@@ -145,19 +166,281 @@ def login():
+        """Check HTTP management accessibility with default password for devices."""
+        
+        for device in devices:
+            ip = device.get("ip")
+            if not ip:
+                continue
+                
+            try:
+                # Use the same method as in async_step_local_web
+                err, token = await self.hass.async_add_executor_job(
+                    SSCPOE_web_login, ip, SSCPOE_LOCAL_DEF_PASSWORD
+                )
+                if not err:
+                    # Add token for further use
+                    device["web_token"] = token
+                else:
+                    LOGGER.debug("SSCPOE HTTP access check failed for %s: %s", ip, err)
+            except Exception as e:
+                LOGGER.debug("SSCPOE HTTP access check exception for %s: %s", ip, e)
+                
+        return None
+    
+    def _get_interface_choices(self) -> list[str]:
+        """Get list of interfaces with their IP addresses, hiding interfaces without IP."""
+        try:
+            interfaces = []
+            for idx, name in socket.if_nameindex():
+                ip = _get_ipv4_for_interface(name)
+                if ip:
+                    interfaces.append(f"{name} ({ip})")
+            return [DEFAULT_BIND_INTERFACE] + sorted(interfaces)
+        except Exception:
+            return [DEFAULT_BIND_INTERFACE]
+
+    async def async_step_local_select(self, user_input: dict[str, str] = None) -> FlowResult:
+        """Screen 3L: confirm/choose local device to add (web/manual or multicast)."""
+        if user_input:
+            action = user_input["action"]
+    
+            if action.startswith("web_"):
+                ip = action[4:]
+    
+                # Otherwise (multicast), go through normal web step with prefilled IP
+                return self.async_show_form(
+                    step_id="web",
+                    data_schema=vol.Schema(
+                        {
+                            vol.Required(CONF_IP_ADDRESS, default=ip): str,
+                            vol.Required(CONF_PASSWORD, default="123456"): str,
+                        }
+                    ),
+                )
+    
+            # old API by SN (or fallback ip)
+            if action.startswith("old_"):
+                sn = action[4:]
+                return self.async_show_form(
+                    step_id="local",
+                    data_schema=vol.Schema(
+                        {
+                            vol.Required(CONF_ID, default=sn): str,
+                            vol.Required(CONF_PASSWORD, default="123456"): str,
+                        }
+                    ),
+                )
+            
+            return self.async_abort(reason="unknown_action")
+    
+        actions = {}
+        for device in (self.local_devices or []):
+            if device.get("add_method") == "old":
+                sn = device.get("sn") or device.get("ip", "unknown")
```

Comment:
I don't see a reason to use IP here. Please replace with
sn = device.get("sn", "unknown")